core/db_models/fields/EE_Datetime_Field.php

```diff
@@ -573,17 +573,11 @@ public function prepare_for_set_from_db($datetime_string)
         if (empty($datetime_string)) {
             $DateTime = new DbSafeDateTime(\EE_Datetime_Field::now, $this->get_UTC_DateTimeZone());
         } else {
-            $DateTime = DateTime::createFromFormat(
+            $DateTime = DbSafeDateTime::createFromFormat(
                 EE_Datetime_Field::mysql_timestamp_format,
                 $datetime_string,
                 $this->get_UTC_DateTimeZone()
             );
-            if ($DateTime instanceof \DateTime) {
-                $DateTime = new DbSafeDateTime(
-                    $DateTime->format(\EE_Datetime_Field::mysql_timestamp_format),
-                    $this->get_UTC_DateTimeZone()
-                );
-            }
```

Comment:
kinda silly we didn't think to do this earlier :\

Reply:
yeah but also understandable.